logstash-core/lib/logstash/output_delegator_strategy_registry.rb

```diff
@@ -0,0 +1,28 @@
+module LogStash; class OutputDelegatorStrategyRegistry
+  # This is generally used as a singleton
+  # Except perhaps during testing
+  def self.instance
+    @instance ||= self.new
```

Comment:
OK, so it looks like this should just proxy through to `@output_class.config_name`, and in other places where we use uniqueness `#id` should be invoked directly. Will push this up in my next commit.